conformance/utils/http/http.go

```diff
@@ -321,62 +325,115 @@ func CompareRoundTrip(t *testing.T, req *roundtripper.Request, cReq *roundtrippe
+					cReq.Headers[strings.ToLower(name)] = val
+				}
+				for name, expectedVal := range expected.ExpectedRequest.Headers {
+					actualVal, ok := cReq.Headers[strings.ToLower(name)]
+					if !ok {
+						return fmt.Errorf("expected %s header to be set, actual headers: %v", name, cReq.Headers)
+					} else if strings.Join(actualVal, ",") != expectedVal {
+						return fmt.Errorf("expected %s header to be set to %s, got %s", name, expectedVal, strings.Join(actualVal, ","))
+					}
 				}
 			}
+
+			// Verify that headers expected *not* to be present on the
+			// request are actually not present.
+			if len(expected.ExpectedRequest.AbsentHeaders) > 0 {
+				for name, val := range cReq.Headers {
+					cReq.Headers[strings.ToLower(name)] = val
+				}
+
+				for _, name := range expected.ExpectedRequest.AbsentHeaders {
+					val, ok := cReq.Headers[strings.ToLower(name)]
+					if ok {
+						return fmt.Errorf("expected %s header to not be set, got %s", name, val)
+					}
+				}
+			}
+
+			if !strings.HasPrefix(cReq.Pod, expected.Backend) {
+				return fmt.Errorf("expected pod name to start with %s, got %s", expected.Backend, cReq.Pod)
+			}
 		}
 
-		if expected.Response.Headers != nil {
+		if expected.Response.Headers != nil || expected.Response.HeadersWithMultipleValues != nil {
 			if cRes.Headers == nil {
-				return fmt.Errorf("no headers captured, expected %v", len(expected.ExpectedRequest.Headers))
+				return fmt.Errorf("no headers captured, expected %v", len(expected.Response.Headers))
 			}
 			for name, val := range cRes.Headers {
 				cRes.Headers[strings.ToLower(name)] = val
 			}
 
-			for name, expectedVal := range expected.Response.Headers {
-				actualVal, ok := cRes.Headers[strings.ToLower(name)]
-				if !ok {
-					return fmt.Errorf("expected %s header to be set, actual headers: %v", name, cRes.Headers)
-				} else if strings.Join(actualVal, ",") != expectedVal {
-					return fmt.Errorf("expected %s header to be set to %s, got %s", name, expectedVal, strings.Join(actualVal, ","))
+			if expected.Response.HeadersWithMultipleValues != nil {
+				for name, expectedVals := range expected.Response.HeadersWithMultipleValues {
+					actualVal, ok := cRes.Headers[strings.ToLower(name)]
+					if !ok {
+						return fmt.Errorf("expected %s header to be set, actual headers: %v", name, cRes.Headers)
+					}
+					actualValStr := strings.Join(actualVal, ",")
+					if expected.Response.IgnoreWhitespace {
+						actualValStr = strings.ReplaceAll(actualValStr, " ", "")
+						for i := range expectedVals {
+							expectedVals[i] = strings.ReplaceAll(expectedVals[i], " ", "")
+						}
+					}
+					matched := false
+					for _, expectedVal := range expectedVals {
+						if actualValStr == expectedVal {
+							matched = true
+							break
+						}
+					}
+					if !matched {
+						return fmt.Errorf("expected %s header to be set to one of %v, got %s", name, expectedVals, actualValStr)
+					}
+				}
+			} else {
```

Comment:
I think this else is unnecessary, right?

Like, you can check the headerswithMultipleValues, and then if Headers is also set you check it.

I think the whole `if expected.Response.HeadersWithMultipleValues != nil {` can be removed, you simply iterate over the HeadersWithMultipleValues first, and then on the Response.Headers, wdyt?